### PR TITLE
[HDF5] Renaming NodalResults to NodalSolutionStepData

### PR DESCRIPTION
**Pull request by sunethwarna**
This PR renames NodalResults to NodalSolutionStepData

### CODE REVIEW

File: applications/HDF5Application/python_scripts/partitioned_single_mesh_primal_input_process.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def Factory(settings, Model):
     settings.ValidateAndAssignDefaults(default_settings)
     model_part = Model[settings["model_part_name"].GetString()]
     hdf5_file_factory = hdf5_io.HDF5ParallelFileFactory(settings["file_settings"])
-    nodal_results_input = hdf5_io.PrimalBossakInput(settings["nodal_results_settings"])
+    nodal_results_input = hdf5_io.PrimalBossakInput(settings["nodal_solution_step_data_settings"])
```
Comment (msandre) [inline applications/HDF5Application/python_scripts/partitioned_single_mesh_primal_input_process.py:23]: Here we can also rename `nodal_results_input` accordingly.

File: applications/HDF5Application/python_scripts/hdf5_io.py [hunk 2]
```diff
@@ -82,7 +82,7 @@ def Execute(self, model_part, hdf5_file):
         KratosHDF5.HDF5ElementSolutionStepDataIO(self.settings, hdf5_file).WriteElementResults(model_part.Elements)
 
 
-class NodalResultsOutput(IOObject):
+class NodalSolutionStepDataOutput(IOObject):
```
Comment (msandre) [inline applications/HDF5Application/python_scripts/hdf5_io.py:85]: Also update NodalResultsInput

**Review by msandre — COMMENTED** [review]
Also change temporal_output_process_factory.py

File: applications/HDF5Application/python_scripts/hdf5_io.py (hunk 2 above)
Comment (sunethwarna) [reply applications/HDF5Application/python_scripts/hdf5_io.py:85]: Done

File: applications/HDF5Application/python_scripts/partitioned_single_mesh_primal_input_process.py (hunk 1 above)
Comment (sunethwarna) [reply applications/HDF5Application/python_scripts/partitioned_single_mesh_primal_input_process.py:23]: Done

**Review by msandre — APPROVED** [review]
Thanks

**Pull request merged**